### PR TITLE
Graded spikes

### PR DESCRIPTION
**Pull request by ahenkes1**
Dear Devs,

I propose the usage of graded spikes instead of "just" binary spikes. I tried to implement the bare infrastructure in the spirit of the existing code base.

Basically, the "fire()" function multiplies the spikes with a value to obtain graded spikes. The factor is learnable, much like, e.g., the threshold value.

During my experiments on the standard snnTorch code with graded spikes I used a nn.Linear() (which is really an affine transformation, not a linear one!) on the output spikes. The current implementation in this PR just uses a linear scaling instead of an affine transformation. 

This is only a first start, but I feel that experienced Devs should take a look at this PR to ensure the foundation of this new feature are settled right from the start to avoid time-consuming refactoring.

Best,
Alex

### CODE REVIEW

**Comment by jeshraghian** [comment]
Awesome, thanks! This looked like a tedious clean-up effort.

Graded spikes buffer has been added. Are you thinking of implementing it as an argument that is embedded within neurons, or as an independent layer that spikes can be passed through (e.g., similar to say, maxpooling or batchnorm).

I kind of think a separate layer would work better for modularity.

**Comment by ahenkes1** [comment]
Yes I can make an independent layer out of it!
